src/common/net.cc

```diff
@@ -171,9 +185,28 @@ namespace Pistache
 
     std::string IP::toString() const
     {
+        if (addr_.ss_family == AF_UNIX)
+        {
+            auto& unAddr = reinterpret_cast<const struct sockaddr_un&>(addr_);
+            if (unAddr.sun_path[0] == '\0')
+            {
+                // The socket is abstract (not present in the file system name
+                // space).  Its name starts with the byte following the initial
+                // NUL.  As the name may contain embedded NUL bytes and its
+                // length is not available here, simply note that it's an
+                // abstract address.
+                return std::string("[Abstract]");
```

Comment:
Supporting Linux abstract sockets doesn't preclude extending the code base to additional platforms.  If one should attempt to create an abstract socket on a platform that doesn't support them, the bind() system call will fail and the error will propagate upward.  But no additional harm is done; it's simply a runtime error.

As a usefulness data point, it turns out that the VMware user world implementation of unix domain sockets treats the pathname variant as if it were abstract.  That is, the implementation doesn't register them in the file system name space.  (This is a security measure of dubious utility to hide sockets from attackers.)

Let me add a request for adding portability to epoll-like interfaces.  It also turns out that VMware's user world emulation of Linux interfaces omits the timerfd_* functions, but does include epoll.  That means that we won't be able to use this patch in our product without persuading VMware to add these functions to their emulation.  (That will be an uphill battle, but it will be greatly aided by this patch being accepted.  In the interim we're forced to use a hacked version of Pistache predating its use of the timerfd_* functions that supports only unix domain sockets.  This situation disgusts us and we want to fix it.)